lib/note-content-editor.jsx

```diff
@@ -171,6 +220,29 @@ export default class NoteContentEditor extends React.Component {
 		)
 	}
 
+	handleReturn = () => {
+		// matches lines that start with `- ` or `* `
+		// preceded by 0 or more tab characters
+		const listItemRe = /^\t*[-*]\s/;
+
+		const { editorState } = this.state;
+		const line = getCurrentBlock( editorState ).getText();
+
+		const trimmedLine = line.trim()
+		if ( trimmedLine === '-' || trimmedLine === '*' ) {
+			this.handleEditorStateChange( finishList( editorState ) );
+			return 'handled'
+		}
+
+		const listItemMatch = line.match( listItemRe )
+		if ( listItemMatch ) {
+			this.handleEditorStateChange( continueList( editorState, listItemMatch ) );
+			return 'handled'
+		}
+
+		return 'not-handled';
```

Comment:
nice @nfcampos - I was unaware of those. I'm a bit nervous about the string values here for reasons like how I just mistook them. I wonder if constant keywords would be better here, something such as `DRAFTJS_HANDLER_CONTINUE` or `DRAFTJS_HANDLER_FINISH` - what do you think?